cwltool/singularity.py

```diff
@@ -143,6 +146,9 @@ def get_image(
         """
         found = False
 
+        if "dockerImageId" not in dockerRequirement and "dockerPull" in dockerRequirement:
+            dockerRequirement["dockerImageId"] = dockerRequirement["dockerPull"]
+
```

Comment:
This is breaking `dockerPull`; if you need this only for `dockerFile`, then set that there

https://github.com/common-workflow-language/cwltool/actions/runs/6947078679/job/19231705078?pr=1938#step:8:165

(missing the `docker://` prefix set at https://github.com/common-workflow-language/cwltool/pull/1938/files#diff-dc5ddffad347b39baf2c837e6f60869433318cc3023bb81bc90e7b480da6e34fR198 because `dockerImageId` was set here)